tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,139 @@
+# SUSE's openQA tests
+#
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Managing Virtual Networks with virsh
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use set_config_as_glue;
+
+sub run {
+
+    #Prepare New Guest Network Interface Configuration
+    foreach my $guest (keys %xen::guests) {
+        assert_script_run("ssh root\@$guest 'cd /etc/sysconfig/network/; cp ifcfg-eth0 ifcfg-eth1'", 60);
+    }
+
+#NAT BASED NETWORK
+#Prepare Test
+my $vnet_nated = <<'EOF';
+<network>
+ <name>vnet_nated</name>
+ <bridge name='virbr1' stp='on' delay='0'/>
+ <forward mode='nat'/>
+ <ip address="192.168.122.1" netmask="255.255.255.0">
+  <dhcp>
+   <range start="192.168.122.2" end="192.168.122.254" />
+  </dhcp>
+ </ip>
+</network>
+EOF
+    foreach my $guest (keys %xen::guests) {
+        record_info "virsh-virtual_network", "NAT BASED NETWORK";
+	assert_script_run("echo '$vnet_nated' >> vnet_nated.xml");
+	assert_script_run("virsh net-create vnet_nated.xml");
+	assert_script_run("virsh net-info vnet_nated");
+	assert_script_run("virsh attach-interface $guest network vnet_nated --live");
+	#my $gi_vnet_nated = script_output("sleep 30;journalctl | grep dnsmasq-dhcp| tail -1| grep -Po "\d+\.\d+\.\d+\.\d+"");
+	assert_script_run("ssh root\@$guest 'ping -c5 -W1 -I eth1 10.160.0.207'",60);
```

Comment:
What if eth1 is missing or bad on host? Please add code in preparation for checking and selecting the network interface for this purpose.

Reply:
Good point, will update code in  preparation for checking and selecting the network interface before Managing Virtual Networks with virsh test